addons/mail/static/src/models/thread/thread.js

```diff
@@ -870,6 +870,18 @@ function factory(dependencies) {
             return [['replace', currentPartnerOrderedSeenMessages.slice().pop()]];
         }
 
+        /**
+         * @private
+         * @returns {mail.message|undefined}
+         */
+        _computeLastMessage() {
+            const {
+                length: l,
+                [l - 1]: lastMessage,
+            } = this.orderedMessages;
```

Comment:
I have learned to know what this means by now but I still find this syntax really unclear.

Reply:
It's equivalent to:
```js
const lastMessage = this.orderedMessages[this.orderedMessages.length - 1];
```
But I don't like this because it repeats the array and inner expression of array index is hard to read. Would be nice if arrays had a `.last` accessor, but it doesn't.

So syntax above is multi-line and use some ES6 stuffs, but at least it's explicit and doesn't repeat array.